firebase/functions/src/index.ts

```diff
@@ -179,6 +205,21 @@ export const createChain = functions
           chainId: chainData.id,
           role: user.customClaims?.role ?? ROLE_CHAINADMIN,
         });
+
+        let name;
+        let email;
+        try {
+          name = user.displayName!;
+          email = user.email!;
+
+          addContactToMailchimpAudience(name, email);
+
+          console.log("Mailchimp add contact success");
+        } catch (error) {
+          console.error("Mailchimp add contact error", email, error);
+        }
```

Comment:
Include the logging in the new function since it's duplicated where you're calling the function